src/main/java/uk/nhs/digital/emailadapter/schedulingtasks/EmailScheduledTask.java

```diff
@@ -0,0 +1,180 @@
+  public EmailScheduledTask(@Value(value = "${mpCryptoPassword}") final String mpCryptoPassword)
+      throws Exception {
+    this.mpCryptoPassword = mpCryptoPassword;
+    decryptor.setPassword(mpCryptoPassword);
+    ExchangeCredentials credentials = new WebCredentials(
+        decryptor.decrypt(getParameter("username")), decryptor.decrypt(getParameter("password")));
+    service.setCredentials(credentials);
+    service.autodiscoverUrl(decryptor.decrypt(getParameter("username")));
+  }
+
+  @Async
+  @Scheduled(fixedRateString = "${fixedRate.in.milliseconds}",
+      initialDelayString = "${initialDelay.in.milliseconds}")
+  public void sendMails() throws InterruptedException {
+    String timeStamp =
+        DateTimeFormatter.ofPattern("yyyy-MM-dd_HH:mm:ss").format(LocalDateTime.now());
+    log.info("Invocation started: {}", timeStamp);
+
+    try {
+      StagedStopwatch stopwatch = StagedStopwatch.start();
+      ItemView view = new ItemView(10);
+      SearchFilterCollection searchFilterCollection =
+          new SearchFilter.SearchFilterCollection(LogicalOperator.And);
+      searchFilterCollection.add(
+          new SearchFilter.IsEqualTo(EmailMessageSchema.From, getParameter("EMS_REPORT_SENDER")));
+      searchFilterCollection.add(new SearchFilter.IsEqualTo(EmailMessageSchema.IsRead, false));
+      FindItemsResults<Item> findResults =
+          service.findItems(WellKnownFolderName.Inbox, searchFilterCollection, view);
+
+      while (findResults.getTotalCount() > 0) {
+        for (Object item : findResults.getItems()) {
+          EmailMessage emailMessage = (EmailMessage) item;
+          emailMessage
+              .load(new PropertySet(BasePropertySet.FirstClassProperties, ItemSchema.MimeContent));
+          log.info("attachment count: {} ", emailMessage.getAttachments().getCount());
+
+          Attachment attachment = emailMessage.getAttachments().getItems().get(0);
+          stopwatch.finishStage("Getting html attchement from email");
+
+          if (attachment instanceof FileAttachment) {
+            FileAttachment fileAttachment = (FileAttachment) attachment;
+            fileAttachment.load();
+
+            // convert bytes[] to string
+            String htmlString = new String(fileAttachment.getContent(), StandardCharsets.UTF_8);
+            byte[] transform = new PDFTransformer().transform(Jsoup.parse(htmlString).html());
+            stopwatch.finishStage("pdf transformation");
+
+            // Create an email message and set properties on the message.
+            EmailMessage message = new EmailMessage(service);
+            message.setSubject(getParameter("EMS_REPORT_SUBJECT"));
+            message.setBody(new MessageBody(getParameter("EMS_REPORT_BODY")));
+            message.getToRecipients().add(getParameter("EMS_REPORT_RECIPIENT"));
+
+            FileAttachment addFileAttachment = message.getAttachments()
+                .addFileAttachment(createFileName(Jsoup.parse(htmlString)), transform);
```

Comment:
refactor and do the below call once rather than multiple times

Jsoup.parse(htmlString)